rotkehlchen/api/rest.py

```diff
@@ -1555,6 +1555,68 @@ def process_history(
         result_dict = _wrap_in_result(result=result, message=msg)
         return api_response(result_dict, status_code=status_code)
 
+    def _process_history_debug(
+        self,
+        from_timestamp: Timestamp,
+        to_timestamp: Timestamp,
+        path: Optional[Path],
+    ) -> Response:
+        error_or_empty, events = self.rotkehlchen.events_historian.get_history(
+            start_ts=from_timestamp,
+            end_ts=to_timestamp,
+            has_premium=self.rotkehlchen.premium is not None,
+        )
+        settings = self.rotkehlchen.get_settings()
+        ignored_ids = self.rotkehlchen.data.db.get_ignored_action_ids(None)
+
+        debug_data = {
+            'events': [entry.serialize() for entry in events],
+            'settings': process_result(settings),
+            'ignored_events_ids': {str(k): v for k, v in ignored_ids.items()},
+            'message': error_or_empty,
+        }
+
+        if path is not None:
+            try:
+                with open(f'{path}/pnl_debug.json', 'w') as f:
+                    json.dump(debug_data, f, indent=4)
+            except (PermissionError) as e:
```

Comment:
For one exception no need for parentheses

```suggestion
            except PermissionError as e:
```